### PR TITLE
QE: Fix test suite issues for build validation

### PR DESCRIPTION
**Pull request by nodeg**
### TODO

- [x] wait for feedback from mc on the `access_compat` removal. mc confirmed that we do not use access_compat anymore. This was something for Apache 2.2 which we do not support since a long time anymore. So it is fine to remove the line from the test.

This will fix test suite issues I found during the 4.3.8 BV

- [x] It seems we do not have the `access_compat` apache module loaded on SUMA 4.3. On Uyuni this module is present.
- [x] Oracle 9 CLM fixes
- [x] Adapt SLE Micro monitoring tests since not all exporters are available, only the node exporter is available: https://bugzilla.suse.com/show_bug.cgi?id=1212248
```salt
----------
          ID: node_exporter
    Function: pkg.installed
        Name: golang-github-prometheus-node_exporter
      Result: true
     Comment: 1 targeted package was installed/updated.
     Started: 03:21:31.704386
    Duration: 4529.007
         SLS: prometheus-exporters
     Changed: golang-github-prometheus-node_exporter:
                  old: ''
                  new: 1.5.0-150100.3.26.1
              
----------
          ID: node_exporter
    Function: file.managed
        Name: /etc/sysconfig/prometheus-node_exporter
      Result: true
     Comment: File /etc/sysconfig/prometheus-node_exporter updated
     Started: 03:21:36.236282
    Duration: 76.993
         SLS: prometheus-exporters
     Changed: diff: "--- 
              +++ 
              @@ -1,11 +1,9 @@
              -## Path:        Network/Monitors/Prometheus/node_exporter
              -## Description: Prometheus node exporter startup parameters
              \
                  -## Type:        string
              -## Default:     ''
              +## Path:           Applications/NodeExporter
              +## Description:    Prometheus exporter for machine metrics
              \
                  +## Type:           string()
              +## Default:        \"\"
              +## ServiceRestart: prometheus-node_exporter
               #
              -# Additional arguments for the node_exporter.
              \
                  -# Please call: /usr/bin/node_exporter --help
              -# for a full list of possible options. 
              -# Note: Please keep the list on one line, of possible.
              +#\
                  \ Arguments for running prometheus-node_exporter
               #
              -ARGS=''
              +ARGS=--collector.systemd --web.listen-address=:9100
              "
              
----------
          ID: node_exporter
    Function: service.running
        Name: prometheus-node_exporter
      Result: true
     Comment: Service restarted
     Started: 03:21:36.987878
    Duration: 14.805
         SLS: prometheus-exporters
     Changed: prometheus-node_exporter: true
              
----------
          ID: apache_exporter
    Function: pkg.installed
        Name: golang-github-lusitaniae-apache_exporter
      Result: false
     Comment: An error was encountered while installing package(s): Zypper command failure: Package 'golang-github-lusitaniae-apache_exporter' not found.Loading repository data...
Reading installed packages...
     Started: 03:21:37.003766
    Duration: 1470.762
         SLS: prometheus-exporters
     Changed: {}
----------
          ID: apache_exporter
    Function: file.managed
        Name: /etc/sysconfig/prometheus-apache_exporter
      Result: false
     Comment: One or more requisite failed: prometheus-exporters.apache_exporter
     Started: 03:21:38.475034
    Duration: 0.003
         SLS: prometheus-exporters
     Changed: {}
----------
          ID: apache_exporter
    Function: service.running
        Name: prometheus-apache_exporter
      Result: false
     Comment: One or more requisite failed: prometheus-exporters.apache_exporter
     Started: 03:21:38.475319
    Duration: 0.002
         SLS: prometheus-exporters
     Changed: {}
----------
          ID: postgres_exporter
    Function: pkg.installed
        Name: prometheus-postgres_exporter
      Result: false
     Comment: An error was encountered while installing package(s): Zypper command failure: Package 'prometheus-postgres_exporter' not found.Loading repository data...
Reading installed packages...
     Started: 03:21:38.475376
    Duration: 1365.875
         SLS: prometheus-exporters
     Changed: {}
----------
          ID: postgres_exporter
    Function: file.managed
        Name: /etc/sysconfig/prometheus-postgres_exporter
      Result: false
     Comment: One or more requisite failed: prometheus-exporters.postgres_exporter
     Started: 03:21:39.841766
    Duration: 0.003
         SLS: prometheus-exporters
     Changed: {}
----------
          ID: postgres_exporter
    Function: service.running
        Name: prometheus-postgres_exporter
      Result: false
     Comment: One or more requisite failed: prometheus-exporters.postgres_exporter
     Started: 03:21:39.842069
    Duration: 0.003
         SLS: prometheus-exporters
     Changed: {}
----------
```
- [x] Exclude SLE Micro SSH minions from package installation/removal tests since it is not supported: https://bugzilla.suse.com/show_bug.cgi?id=1208045

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Cucumber tests were added

- [x] **DONE**
## Links

- Manager 4.3: 
- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by ktsamis — APPROVED** [review]

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (vandabarata) [inline testsuite/features/build_validation/smoke_tests/smoke_tests.template:164]:
```suggestion
    And I check "node" exporter
    And I click on "Save"
    Then I should see a "Formula saved" text
```

Comment (vandabarata) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template:164]: you're missing saving the formula before checking if it's been saved

Comment (nodeg) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template:164]: Nice catch. Thank you!

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (vandabarata) [inline testsuite/features/build_validation/smoke_tests/smoke_tests.template:187]: does it need to be explicitly started and enabled? shouldn't the highstate be taking care of that?

Comment (nodeg) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template:187]: Yes, it must be handled manually because of https://bugzilla.suse.com/show_bug.cgi?id=1209374.

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (nodeg) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template:164]: Fixed.

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (vandabarata) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template]: this sounds like a step only for sle micro then - can you separate it into a different scenario just for SLE Micro? The other systems should work without this with just the Highstate

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (vandabarata) [inline testsuite/features/build_validation/smoke_tests/smoke_tests.template:197]:
```suggestion
    When I wait until "apache" exporter service is active on "<client>"
```

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (nodeg) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template]: Yes, that is SLE Micro specific. In BV that is sadly not that easy with the way the test suite report works. As you can see, we nearly always use tags like `@skip_for_*`. @Bischoff can tell you more about that.

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (nodeg) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template]: Fixed.

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (vandabarata) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template]: what if instead of "client" we substitute it for SLE Micro, in these 2 steps? :thinking: that way, it won't apply to the other clients

Comment (nodeg) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template]: This would work, but it would still show up in the test suite report for other non SLE Micro clients IIRC. I sadly have no example or screenshot to demonstrate that.

Comment (vandabarata) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template]: it's fine for me if it shows up, as long as it doesn't execute. I just want to be sure the other clients are being tested appropriately, where the highstate should be enough to take care of the services being started and enabled. you can leave a comment before those 2 steps specifying the bug you linked with a TODO or WORKAROUND so we know to take it out in the future

Comment (vandabarata) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template]: that would work for me :thinking: @Bischoff @ktsamis what do you think?

Comment (nodeg) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template]: https://github.com/uyuni-project/uyuni/pull/7525/commits/2898557e33db993703223184ee9d82336eb26431

Comment (ktsamis) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template]: Yes this is the better, cleaner way to handle this and I especially like the workaround note. The separation of this with the tag is the cleaner way to do this so we don't hide any future bugs accidentally. Thanks for this :)

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (nodeg) [reply testsuite/features/build_validation/smoke_tests/smoke_tests.template:191]: Fixed.

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:693]: To me the full sentence looks more natural without "the".

File: testsuite/features/build_validation/init_clients/proxy.feature
Comment (srbarrios) [inline testsuite/features/build_validation/init_clients/proxy.feature:55]: Please if this step is a valid check in Uyuni, let's differ on the scenario using a label.

Comment (srbarrios) [inline testsuite/features/build_validation/init_clients/proxy.feature:55]: Did you discuss with dev team the fact that we don't have access_compat in SUMA?
This difference is expected from a design point of view?

File: testsuite/features/support/env.rb
Comment (srbarrios) [inline testsuite/features/support/env.rb:488]:
```suggestion Before('@skip_for_sle_micro_ssh_minion') do |scenario|
   sle_micro_ssh_nodes = %w[slemicro51_ssh_minion slemicro52_ssh_minion slemicro53_ssh_minion slemicro54_ssh_minion]
   current_feature_node = scenario.location.file.split(%r{(\.feature|\/)})[-2] # We can extract this in a method, i.e. get_feature_name(scenario)
   skip_this_scenario if sle_micro_ssh_nodes.include? current_feature_node
 end
```

**Review by srbarrios — COMMENTED** [review]
I would specially review the access_compat diversion before merge this PR.

File: testsuite/features/build_validation/init_clients/proxy.feature
Comment (nodeg) [reply testsuite/features/build_validation/init_clients/proxy.feature:55]: I will recheck the spec file and adjust the code afterwards.

Comment (nodeg) [reply testsuite/features/build_validation/init_clients/proxy.feature:55]: Not yet. I will ask in our team channel and report back.

File: testsuite/features/step_definitions/command_steps.rb
Comment (nodeg) [reply testsuite/features/step_definitions/command_steps.rb:693]: Example: `I start apache service` vs. `I start the apache service`, where the latter one sounds more grammatically correct to me.

File: testsuite/features/build_validation/init_clients/proxy.feature
Comment (nodeg) [reply testsuite/features/build_validation/init_clients/proxy.feature:55]: It seems, in SUMA, we explicitly removed this. See https://github.com/SUSE/spacewalk/commit/73df9feac4563c6d6c1c2f45b279ae02d263baab
I can also no longer find it in the `spacewalk-proxy.spec` file on Uyuni. No idea what happened and why I saw it when I implemented the check.
We also removed it in Uyuni in 2019 according to the changelog: https://github.com/uyuni-project/uyuni/blob/089407703cafdc4b0a09659aadc53d42c53d8ac9/proxy/proxy/spacewalk-proxy.changes#L207

File: testsuite/features/support/env.rb
Comment (nodeg) [reply testsuite/features/support/env.rb:488]: This does not work:

```ruby
Before('@skip_for_sle_micro_ssh_minion') do |scenario|
  sle_micro_ssh_nodes = %w[slemicro51_ssh_minion slemicro52_ssh_minion slemicro53_ssh_minion slemicro54_ssh_minion]
  current_feature_node = scenario.location.file.split(%r{(\.feature|\/)})[-2] # We can extract this in a method, i.e. get_feature_name(scenario)
  puts sle_micro_ssh_nodes.include? current_feature_node
  puts current_feature_node
  skip_this_scenario if sle_micro_ssh_nodes.include? current_feature_node
end
```

 ```cucumber
Feature: doms tests

@skip_for_sle_micro_ssh_minion
  Scenario: doms tests
    When I run doms tests
```

```bash
uyuni-bv-master-ctl:~/spacewalk/testsuite # cucumber features/slemicro51_ssh_minion_smoke_tests.feature 
(...)
Initializing a twopence node for 'server'.
Host 'server' is alive with determined hostname uyuni-bv-master-srv and FQDN uyuni-bv-master-srv.mgr.suse.de
Node: uyuni-bv-master-srv, OS Version: 15.4, Family: opensuse-leap
Activating HTTP API
Using the default profile...
Feature: doms tests

  @skip_for_sle_micro_ssh_minion
  Scenario: doms tests    # features/slemicro51_ssh_minion_smoke_tests.feature:4
      This scenario ran at: 2023-10-02 10:59:12 +0200
false
slemicro51_ssh_minion_smoke_tests
(...)
    When I run doms tests # features/step_definitions/common_steps.rb:536
      This scenario took: 1 seconds

1 scenario (1 passed)
1 step (1 passed)
0m0.589s
```
It should at least return true.

File: testsuite/features/build_validation/init_clients/proxy.feature
Comment (nodeg) [reply testsuite/features/build_validation/init_clients/proxy.feature:55]: We will need to wait for feedback from mc once he is back from vacation.

Comment (nodeg) [reply testsuite/features/build_validation/init_clients/proxy.feature:55]: Ok, mc confirmed that we do not use `access_compat` anymore. This was something for Apache 2.2 which we do not support since a long time anymore. So it is fine to remove the line from the test.

Comment (nodeg) [reply testsuite/features/build_validation/init_clients/proxy.feature:55]: See explanation below.

File: testsuite/features/support/env.rb
Comment (srbarrios) [reply testsuite/features/support/env.rb:488]: Sure thing, we will need to remove "_smoke_tests" too :) I fixed my suggestion above.

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:693]: Yeah might be me badly translating it from spanish :D Nevermind :) "I start Apache's service" is how I would translated it, but looking other parts of our code, yeah, make sense to add the "the" I guess.

File: testsuite/features/support/env.rb
Comment (nodeg) [reply testsuite/features/support/env.rb:488]: I can confirm that this does work now:

```bash
uyuni-bv-master-ctl:~/spacewalk/testsuite # cucumber features/slemicro51_ssh_minion_smoke_tests.feature
(...)
Capybara APP Host: https://uyuni-bv-master-srv.mgr.suse.de:8888
Initializing a twopence node for 'server'.
Host 'server' is alive with determined hostname uyuni-bv-master-srv and FQDN uyuni-bv-master-srv.mgr.suse.de
Node: uyuni-bv-master-srv, OS Version: 15.4, Family: opensuse-leap
Activating HTTP API
Using the default profile...
Feature: doms tests

  @skip_for_sle_micro_ssh_minion
  Scenario: doms tests    # features/slemicro51_ssh_minion_smoke_tests.feature:4
      This scenario ran at: 2023-10-05 16:49:13 +0200
true
slemicro51_ssh_minion
    When I run doms tests # features/step_definitions/common_steps.rb:536
      This scenario took: 0 seconds

1 scenario (1 skipped)
1 step (1 skipped)
0m0.004s
```

**Review by srbarrios — APPROVED** [review]

**Comment by nodeg** [comment]
All checks are green 
image
I will do a rebase to sort my commits and then merge the PR without waiting for the tests to run again.

**Pull request merged**
